frontend/catalyst/debug/compiler_functions.py

```diff
@@ -335,6 +335,7 @@ def f(x):
         f"-Wl,-rpath,{python_lib_dir_path}",
         f"-L{python_lib_dir_path}",
         "-lpython" + version_str,
+        "-lcurses",
```

Comment:
This library is required after we apply LLVM patches. Otherwise we get this error.

```shell
llvm-project/build/lib/libmlir_async_runtime.so.19.0git: undefined reference to `setupterm@NCURSES6_TINFO_5.0.19991023'
```